pyccel/ast/core.py

```diff
@@ -4852,12 +4852,12 @@ def __init__(
             for a,s in zip(args, shape):
                 if isinstance(a, Slice):
```

Comment:
I know this is beyond your PR, but you already did changes in this file so can you update the doc-string of the `IndexedElement` class, the bug is already fixed, and the example does not work well.
so I suggest this example.
```python
"""
    Represents a mathematical object with indices.
    Examples
    --------
    >>> from sympy import symbols, Idx
    >>> from pyccel.ast.core import IndexedVariable, IndexedElement
    >>> i, j = symbols('i j', cls=Idx)
    >>> A = IndexedVariable('A', dtype='int')
    >>> IndexedElement(A, i, j)
    IndexedElement(A, i, j)
    >>> IndexedElement(A, i, j) == A[i, j]
    True
    """
```

Reply:
Sorry I didn't change anything in this class. I won't remove a TODO or a bug comment that I didn't do or fixed and I don't have a slice idea what is about.

Reply:
I think I fixed that bug a while ago. I didn't notice that it was in the docstring. Sorry

I agree that it has nothing to do with this PR, but it's a quick copy and paste and I'm not sure it's worth creating a separate PR just for this. Better to fix it while it's been spotted. I am happy to vouch for the fact that it is fixed.